### PR TITLE
[Bookaroom 2] - Amenities and additional tests

### PR DESCRIPTION
**Pull request by Dairai**
- Added support for room amenities
- Added tests for email validator and TimeParser

### CODE REVIEW

File: src/com/comeet/data/DataRepository.java [hunk 1]
```diff
@@ -65,36 +66,9 @@ private void closeConnection() throws SQLException {
...
-                roomMetadata.setAddress(rs.getString(rs.findColumn("address")));
-                roomMetadata.setNavigationMap(rs.getString(rs.findColumn("navigation")));
-                roomMetadata.setCapacity(rs.getInt(rs.findColumn("Longitude")));
-                roomMetadata.setRoomPic(rs.getString(rs.findColumn("roomPic")));
-                roomMetadata.setState(rs.getString(rs.findColumn("state")));
-            }
-        }
-
-        closeConnection();
+        roomMetadata = getRoomMetadata(email);
```
Comment (kb000) [inline src/com/comeet/data/DataRepository.java]: Declare and assign on the same line

File: src/com/comeet/utilities/Validator.java [hunk 2]
```diff
@@ -6,7 +6,110 @@
 public class Validator {
     
     private static final Pattern VALID_EMAIL_ADDRESS_REGEX = 
-                    Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+                    Pattern.compile("(?:(?:\\r\\n)?[ \\t])*(?:(?:(?:[^()<>@,;:\\\\\".\\[\\] \\000-\\031]+(?:(?:(?:\\r\\n)?[ "
```
Comment (kb000) [inline src/com/comeet/utilities/Validator.java]: Hoooooootttt damn.  Can you putt a comment on this with the source?

File: test/com/comeet/utilities/TimeParseTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    TimeParse parseTime;
+    
+    /**
+     * Test to verify that the date follows ISO format
+     * @throws Exception
+     */
+    @Test(expected=Exception.class)
+    public void TimeParseISOTest() throws Exception
+    {
+        String startDate = DateTime.now().toString();
```
Comment (kb000) [inline test/com/comeet/utilities/TimeParseTest.java]: Does this definitely give back 2017-04-18T21:25:00-0500?

**Review by kb000 — APPROVED** [review]
Approving pending response to feedback.

File: src/com/comeet/utilities/Validator.java (hunk 2 above)
Comment (Dairai) [reply src/com/comeet/utilities/Validator.java]: Will do

File: test/com/comeet/utilities/TimeParseTest.java (hunk 3 above)
Comment (Dairai) [reply test/com/comeet/utilities/TimeParseTest.java]: Oh no. I missed a call to convert to ISO. I will update that

**Pull request merged**